lms/views/analytics.py

```diff
@@ -0,0 +1,110 @@
+from pyramid.view import view_config, view_defaults
+from lms.security import Permissions
+from pyramid.httpexceptions import HTTPFound
+from pyramid.security import remember
+from lms.validation.authentication import BearerTokenSchema
+
+
+@view_config(
```

Comment:
I wanted to try all of these so I made some changes to avoid writing JS I don't know how to write (or not know how to write in 5 minutes) which might make understanding it tricker.


The real process should be, IMO:

- Launch provides API info like it does for other API endpoints.

- The FE should display (somewhere) an entry point for the analysis and use the new API info to submit a form to this endpoint (`lti.analytics.assignment`)

This endpoint just promotes the form token to a cookie and redirects to `analytics.assignment`.


- `analytics.assignment` endpoint should just kick in a new FE miniapp which its own version of JSCOnfig. That JS config contains another API info structure pointing to the  async endpoint (`analytics.assignment.api` to actually get the analytics. 


`analytics.assignment` is the only endpoint that actually reads the cookie. The cookie provides authentication only. Authorization to fetch the analytics is based on the token's identity checked against the DB state. That happens on `analytics.assignment.api`.


- analytics.assignment.api Check authorization and would get the right numbers from H and returns some JSON.